### PR TITLE
Rca toolkit single page experience

### PR DESCRIPTION
**Pull request by h-mangipudi**
## Overview
This tool uses OpenAI/ChatGPT and provides Copilot capabilities to help users generate RCAs for their ICMS/incidents that need to be solved. It serves pre-canned templates and also is able to generate RCA's for custom issues/icms/incidents that user may need an RCA for. This change allows users to quickly retrieve RCA's and have custom RCA's generated for them thus saving them immense time and effort when solving and investigating ICMS and similar issues.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

### This PR:
This PR has created a new component called communication-toolkit and uses open-ai-chat-container component in its html. It also adds the capability to have another row of boxes/tiles for copilot prompts by modifying the chat-ui component. The backend is modified to have the system prompt for the rca copilot. This tool uses GPT4 and WebSocket Protocol. 

## How Can This Be Tested? <span>&#128269;</span>
This can be tested by clicking on the starter prompts of the SuperEasyRCA and also writing own custom prompts to see if well-written robust RCAs are written. 

## Checklist <span>&#9989;</span>
- [ ] I have looked over the diffs.
- [ ] I have run the linter to catch any errors.
- [ ] There are no errors from my changes on this branch.
- [ ] I have ensured that my changes support accessibility and can be accessed with the keyboard alone
- [ ] I have requested review on this PR.

## Screenshots After Fix (if appropriate):

image

image

### CODE REVIEW

File: AngularApp/projects/applens/src/app/modules/dashboard/communication-toolkit/communication-toolkit.component.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      [chatQuerySamplesFileUri]="chatQuerySamplesFileURIPath"
+      [showFeedbackOptions]="true"
+      [showContentDisclaimer]="true"
+      [contentDisclaimerMessage]="contentDisclaimerMessage"
+      [quotaEnforced]="true"
+      [dailyMessageQuota]="20"
+      [messageQuotaWarningThreshold]="10"
+      [chatHeader]= "chatHeader"
+      [systemPhotoSource]="'/assets/img/openailogo.svg'"
+      [chatIdentifier]="'rcacopilot'"
```
Comment (SharmaAjay19) [inline AngularApp/projects/applens/src/app/modules/dashboard/communication-toolkit/communication-toolkit.component.html:17]: This input seems to be provided twice. First time the value is EasyRCAChatGPT and second time its rcacopilot. Maybe standardize the chatIdentifier as rcacopilot and use that for all names.

**Review by SharmaAjay19 — APPROVED** [review]
Wow! Most impact in the least lines of code 👍

File: AngularApp/projects/applens/src/app/modules/dashboard/side-nav/side-nav.component.ts [hunk 2]
```diff
@@ -213,7 +213,21 @@ export class SideNavComponent implements OnInit {
         return this.currentRoutePath && this.currentRoutePath.join('/').toLowerCase() === `kustogpt`;
       },
       icon: null
+    },
+    {
+      label: 'RCA Toolkit',
```
Comment (rekhaswaminathan) [inline AngularApp/projects/applens/src/app/modules/dashboard/side-nav/side-nav.component.ts:218]: I believe we wanted to show this only for Microsoft.Web resources? If so there should be a condition to check that.

File: AngularApp/projects/applens/src/app/modules/dashboard/communication-toolkit/communication-toolkit.component.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+@Component({
+  selector: 'communication-toolkit',
+  templateUrl: './communication-toolkit.component.html',
+  styleUrls: ['./communication-toolkit.component.scss']
+})
+
+export class CommunicationToolkitComponent implements OnInit {
+  @ViewChild('chatbot') OpenAICHATComponentRef: OpenAIChatComponent;
+
+
```
Comment (ShekharGupta1988) [inline AngularApp/projects/applens/src/app/modules/dashboard/communication-toolkit/communication-toolkit.component.ts:23]: nitpick: remove empty lines

**Review by ShekharGupta1988 — APPROVED** [review]

**Review by ShekharGupta1988 — APPROVED** [review]

**Pull request merged**